vivisect/qt/funcgraph.py

```diff
@@ -236,6 +247,55 @@ def __init__(self, vw, vwqgui):
         self.addHotKeyTarget('funcgraph:paintdown', self._hotkey_paintDown)
         self.addHotKey('ctrl+m', 'funcgraph:paintmerge')
         self.addHotKeyTarget('funcgraph:paintmerge', self._hotkey_paintMerge)
+        self.addHotKey('x', 'viv:xrefsto')
+        self.addHotKeyTarget('viv:xrefsto', self._viv_xrefsto)
+
+    def toggleAutoRefresh(self):
+        '''
+        This allows the function graph to auto-refresh on changes (or not)
+        '''
+        self._autorefresh.setChecked(not self._autorefresh.isChecked())
+
+    def _canHazAutoRefresh(self):
+        '''
+        Should we autorefresh on updates?
+        '''
+        return self._autorefresh.isChecked()
+
+    def rendToolsSetName(self):
+        mwname, ok = QInputDialog.getText(self, 'Set Mem Window Name', 'Name')
+        if ok:
+            self.setMemWindowName(str(mwname))
+
+    def getRendToolsMenu(self):
+        menu = e_mem_qt.VQMemoryWindow.getRendToolsMenu(self)
```

Comment:
So, `VQVivFuncgraphView` isn't descended from `VQMemoryWindow`, so we're kinda playing fast and loose here with the type system. Too fast and loose, because this adds a "locked" option to the opts menu that ends up crashing the entire viv process.

Reply:
agreed.  i'm open to guidance on this one.  does it make sense to create a base-class that both VQMemoryWindow and VQVivFuncGraphView derive from?
or should we just generate the menu here from scratch?

Reply:
Hmm. It depends. `VQVivFuncgraphView` and `VQMemoryWindow` are derived from very similar objects, but end up doing very different things and don't share a whole lot else in common.

I do like the idea of a base Vivisect UI class though, but I feel like that is out of scope for here (and deserves its own PR).

Reply:
agreed